build-prep.sh

```diff
@@ -26,26 +27,55 @@ if [ "${USE_MVN}" = 'true' ]; then
         --quiet \
         -DrepoUrl=https://artifacts.alfresco.com/nexus/content/repositories/oracle-java \
         -Dartifact="${JAVA_OS_ARCH}":"${JAVA_SE_TYPE}":"${JAVA_VERSION}":"${JAVA_PACKAGING}":bin \
-        -DoutputDirectory=.
+        -DoutputDirectory="${DOWNLOAD_DIR}"
 
     # Our filenames are munged into Maven compatible names
-    JRE_FILENAME="${JAVA_SE_TYPE}"-"${JAVA_VERSION}"-bin."${JAVA_PACKAGING}"
-else
+    JRE_FILENAME="${DOWNLOAD_DIR}/${JAVA_SE_TYPE}-${JAVA_VERSION}-bin.${JAVA_PACKAGING}"
+}
 
+java::download::oracle () {
     # You can still download the latest Java version from Oracle
 
+    declare OLD_PWD="${PWD}"
+    cd "${DOWNLOAD_DIR}"
     curl -jksSLOH "Cookie: oraclelicense=accept-securebackup-cookie" "${JRE_URL}"
```

Comment:
What happens if `curl` doesn't manage to download the file? Are there any checks for file existence thereafter?

Reply:
The `errexit` line should cause the whole script to fail in that case.

Reply:
Unlike most things I write, this has `set -e` at the top, so the script would exit immediately. The `sha256sum` checks below would catch anything else.

Reply:
Cool. I should've paid more attention to the beginning of the script. :-)